package.json

```diff
@@ -36,21 +37,53 @@
       ],
       "configuration": "./language-configuration.json"
     }],
-    "configurationDefaults": {
-      "[lc2k]": {
-        "editor.insertSpaces": false,
-        "editor.tabSize": 8,
-        "editor.autoIndent": false,
-        "editor.formatOnSave": false
-      }
-    },
+
     "grammars": [{
       "language": "lc2k",
       "scopeName": "source.lc2k",
       "path": "./syntaxes/lc2k.tmLanguage.json"
     }],
     "capabilities": {
       "definitionProvider": "true"
+    },
+    "configuration": {
+      "type": "object",
+      "title": "LC2K configuration",
+      "properties": {
+        "lc2k.format.enable": {
+          "type": "boolean",
+          "default": true,
+          "description": "Enables this extention to format .as files."
+        },
+        "lc2k.format.addNewlineAtEOF": {
```

Comment:
This could probably be replaced by adding to the language-specific settings both:
```json
  "files.trimFinalNewlines": true,
  "files.insertFinalNewline": true,
```

Reply:
did this! now I just have to change the readme